assets/section-image-banner.css

```diff
@@ -220,9 +217,7 @@
     box-shadow: none;
   }
 
-  .banner:not(.banner--mobile-bottom) .button--secondary {
-    --color-button: 255, 255, 255;
```

Comment:
> This should be fine for new merchants, but if a merchant upgrades to this change automatically, they may lose their previously-white text. Is there anything we can do about that?

Unless I'm thinking about this wrong, even if we added a migration to automatically update a theme to, say, inverse scheme, we can't be sure that scheme actually is using white (or even a relatively accessible color). Though this will be a manual update I'm sure so it shouldn't be "automatic" in that way at least, but still breaking.

Reply:
that's a good call 👆

Reply:
Yeah, this is the tricky bit. I'm not sure how, but we might consider keeping the CSS in place for existing merchants somehow.

Reply:
Could we do the reverse? Add a setting like "Use white text and buttons when container is not displayed." We'd want it unchecked by default, but checked automatically when folks upgrade?

It's not all that different, but I think it reinforces that the default (expected) behavior would be that their color schemes always work, and that checking a box would override that.

Reply:
Sounds like a potential plan. But we still need to keep in mind that if we are going to force i.e `inverse` color scheme, it may not have white color for text/button if it was changed by a merchant previously.

Reply:
If we do this plan, any existing merchants who upgrade don't need their color scheme changed at all, right? Since they'll have the box checked, and will keep the forced white color scheme. And any new merchants will just have Dawn's default schemes, so inverse should work fine. Or am I missing a detail?

Reply:
Part of me actually likes that adding a setting would keep the shortcut to white text around. It wasn't perfect logic or UX but the original decision to use white still has _some_ merit in a lot of cases. Not to say I at all love the idea of having to add a setting for essentially deprecated functionally, of course 😅 

If we go this route we should seek opinions about the wording. I wish there was a concise way to kind of hint at why you might ever use such a setting. Not a real recommendation but, you know something like.. `[ ] Override color scheme with white for maximum contrast`

Reply:
> We'd want it unchecked by default, but checked automatically when folks upgrade?

Hmm.. this will all just be thinking aloud more than a recommendation, but in the case of it being a white text setting that defaults to `false`, _I think_ we'd need a migration to add the new setting entry with the desired value (`true`) to upgrading themes.

If the schema default for the new setting was `true` however..
- Upgrading shops would automatically get the correct result because of the absence of the setting in their templates
- New theme installs could receive preset sections with the setting `false` if we add them, but..
- Any newly added (non-preset) sections would still fallback the schema default (`true`)
- In a subsequent release maybe, finally change the schema default to `false`

Does that train of thought make sense?

Reply:
It does. The key objective here is that we want new merchants to have their color scheme text used by default. It sounds like with that plan, we'd get there, but it would take a couple releases. If it's possible to get there in a single release, that would be preferable. But I'd totally defer to someone with more knowledge of our upgrade/migration capabilities — I'm not sure what we can and can't do.

Reply:
Just to close the loop here officially, we're ok with this being a visual change on merchant storefronts since they'll have a chance to review. Going to mark this one as resolved as a result.